.github/templates/main.yml

```diff
@@ -72,6 +73,37 @@ steps:
       retention-days: 1
 #@ end
 
+#@ def buildPackages():
+#@ for pkgName in nugetPackages:
+#@ buildCmd = "MsBuild.exe ${{ github.workspace }}/Realm/" + pkgName + " -t:Pack -p:Configuration=" + configuration + " -p:PackageOutputPath=${{ github.workspace }}/Realm/packages -p:VersionSuffix=${{ steps.set-version-suffix.outputs.build_suffix }} -restore"
+  - name: #@ "Build nuget package " + pkgName
+    run: #@ buildCmd
+#@ end
+#@ end
+
+#@ def uploadPackageArtifacts():
+#@ for pkgName in nugetPackages:
+#@ finalPkgName = pkgName + ".${{ steps.find-nupkg-version.outputs.pacakge_version }}.nupkg"
+  - name: #@ "Store artifacts for " + pkgName
+    uses: actions/upload-artifact@v2
+    with:
+      name: #@ finalPkgName
+      path: #@ "${{ github.workspace }}/Realm/packages/" + finalPkgName
+      retention-days: ${{ github.event_name != 'pull_request' && 30 || 1 }}
+#@ end
+#@ end
+
+#@ def findPackageVersion():
+  - name: Find nupkg version
```

Comment:
[nit] This can return the node directly rather than array element, similar to `checkoutCode` - then you won't need `template.replace` in the steps array.